pkg/login/social/generic_oauth.go

```diff
@@ -239,6 +249,26 @@ func (s *SocialGenericOAuth) extractRole(data *UserInfoJson) (string, error) {
 	return role, nil
 }
 
+func (s *SocialGenericOAuth) extractGroupMappings(data *UserInfoJson) ([]setting.OAuthGroupMapping, error) {
+	var groupMappings []setting.OAuthGroupMapping
+
+	for _, mapping := range s.groupMappings {
+		role, err := s.searchJSONForAttr(mapping.RoleAttributePath, data.rawJSON)
+		if err != nil {
+			s.log.Error("Failed to get role_attribute_path", "error", err)
+			continue
+		}
+		if role == "" {
+			s.log.Debug(fmt.Sprintf("role_attribute_path did not produce a role: %s", mapping.RoleAttributePath))
```

Comment:
Quoting the value is preferable since it may contain whitespace:
```suggestion
			s.log.Debug(fmt.Sprintf("role_attribute_path did not produce a role: %q", mapping.RoleAttributePath))
```